pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -0,0 +1,610 @@
+	}
+
+	// Provision secrets and the trusted bundle into the cluster.
+	hdler := utils.NewComponentHandler(reqLogger, r.client, r.scheme, ls)
+
+	// Before we can create secrets, we need to ensure the tigera-elasticsearch namespace exists.
+	esNamespace := render.CreateNamespace(render.ElasticsearchNamespace, install.KubernetesProvider, render.PSSPrivileged)
+	if err = hdler.CreateOrUpdateOrDelete(ctx, render.NewPassthrough(esNamespace), r.status); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Create Elasticsearch secrets.
+	esTrustedBundle := elasticKeys.trustedBundle(clusterCM)
+	if err = hdler.CreateOrUpdateOrDelete(ctx, elasticKeys.component(esTrustedBundle), r.status); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	if kibanaEnabled {
+		// Create the Namespace.
+		kbNamespace := render.CreateNamespace(render.KibanaNamespace, install.KubernetesProvider, render.PSSPrivileged)
+		if err = hdler.CreateOrUpdateOrDelete(ctx, render.NewPassthrough(kbNamespace), r.status); err != nil {
+			r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+
+		// Render the key pair and trusted bundle into the Kibana namespace.
+		if err = hdler.CreateOrUpdateOrDelete(ctx, elasticKeys.kibanaComponent(esTrustedBundle), r.status); err != nil {
+			r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Generate keys for Tigera components if we're running in multi-tenant mode and this is a reconcile
+	// for a particular tenant, or if not in multi-tenant mode.
+	reconcileTigeraSecrets := !r.multiTenant || r.multiTenant && tenant != nil
+	if !reconcileTigeraSecrets {
+		reqLogger.Info("Skipping render of secrets for Tigera ES components")
+		return reconcile.Result{}, nil
+	}
+	reqLogger.Info("Rendering secrets for Tigera ES components")
+
+	// Create secrets for Tigera components.
+	keyPairs, err := r.generateNamespacedSecrets(reqLogger, helper, appCM)
+	if err != nil {
+		// Status manager is handled already, so we can just return
+		return reconcile.Result{}, err
+	}
+	if keyPairs == nil {
+		// Waiting for keys to be ready.
+		reqLogger.Info("Waiting for key pairs to be ready")
+		return reconcile.Result{}, nil
+	}
+	trustedBundle := keyPairs.trustedBundle(appCM)
+
+	if err = hdler.CreateOrUpdateOrDelete(ctx, keyPairs.component(trustedBundle, helper), r.status); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+	r.status.ClearDegraded()
+	return reconcile.Result{RequeueAfter: 60 * time.Second}, nil
```

Comment:
> and currently at 1 minute it seems like a level that we expect the requeue to be needed for normal operation.

@tmjd This particular rhetoric doesn't makes sense to me. 1 minute isn't any more or less "for normal operation" than 5 or 30 minutes is. The value here should suggest only one thing - how quickly do we want to react in the face of a bug?

So, instead of debating magic numbers, I think we have a few questions to answer first:

- Do we care about this case at all? If not, we just shouldn't requeue.
- If we do care, what is a reasonable amount of time that we can retry that will actually mitigate cluster issues?
- What value is too small, and what is the reason it is too small?

From there, assuming we want to keep this at all, we should pick the smallest value for this number that we possibly can without incurring unreasonable downsides.

Reply:
I think I am in favor of simply removing RequeueAfter altogether, given the above.

Reply:
I am good with removing retries. The only reason in my mind is to ensure that any resources that operator creates, that we don't watch, will be corrected eventually if the user changes them.

To clarify my "normal operation" comment. When I said that I meant as part of normal cluster setup/installation. What that excludes is when a user (or something) modifies a resource that the operator deploys, sorry for not being clear with what I meant.